File: T_Primes.java
import java.util.Scanner;

public class T_Primes {
    static int[] isPrime = new int[1000007];
    static void sieve(){
        isPrime[0]=0;
        isPrime[1]=0;
        for (int i = 2; i < 1000007; i++)
        {
            isPrime[i]=1;
        }
        for (int i = 2; i < 1000007; i++)
        {
            if(isPrime[i]==0){
                for (int j = 2*i; j < 1000007; j+=i)
                {
                    isPrime[j]=0;
                }
            }
        }

    }

    public static void main(String[] args) {
        sieve();
        Scanner scanner=new Scanner(System.in);
        int n=scanner.nextInt();
        for (int i = 0; i < n; i++) {
            long x=scanner.nextLong();
            int root=(int) Math.sqrt(x);
            if((long)root*root==x && isPrime[root]==1 ) System.out.println("YES");
            else System.out.println("NO");
        }
		scanner.close();
    }
}

File: Tprimes.cpp
#include<iostream>
#include<cmath>
using namespace std;

bool is_prime(long long n) {
    if (n < 2) return false;
    for (long long i = 2; i * i <= n; i++) {
        if (n % i == 0) return false;
    }
    return true;
}

int main()
{
    int t;
    cin >> t;
    while(t-- > 0)
    {
        long long x;
        cin >> x;
        long long root = (long long)sqrt(x);
        if (root * root == x && is_prime(root)) cout << "YES" << endl;
        else cout << "NO" << endl;
    }
    return 0;
}
/* AI
#include<iostream>
#include<cmath>
#include<vector>
using namespace std;

const int MAX = 1000001;
vector<bool> is_prime(MAX, true);

void sieve() {
    is_prime[0] = is_prime[1] = false;
    for (int i = 2; i * i <= MAX; i++) {
        if (is_prime[i]) {
            for (int j = i * i; j <= MAX; j += i) {
                is_prime[j] = false;
            }
        }
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    sieve();  // precompute primes
    
    int n;
    cin >> n;
    while(n--) {
        long long x;
        cin >> x;
        long long root = sqrt(x);
        if (root * root == x && root <= 1000000 && is_prime[root])
            cout << "YES\n";
        else
            cout << "NO\n";
    }
    return 0;
}*/